src/kbmod/standardizers/standardizer.py

```diff
@@ -0,0 +1,591 @@
+"""`Standardizer` converts data from a given data source into a `LayeredImage`
+object, applying data-source specific, transformations in the process. A
+layered image is a collection containing:
+ * science image
+ * variance image
+ * mask image
+along with the:
+ * on-sky coordinates of the pixel in the center of the science image
+ * observation timestamp
+ * location of the data source
+
+When possible, standardizers should attempt to extract a valid WCS and/or
+bounding box from the data source.
+"""
+import abc
+import warnings
+
+
+__all__ = ["Standardizer", "StandardizerConfig", "ConfigurationError"]
+
+
+class ConfigurationError(Exception):
+    """Error that is raised when configuration parameters contain a logical error."""
+
+    pass
+
+
+class StandardizerConfig:
+    """Base class for Standardizer configuration.
+
+    Not all standardizers will (can) use the same parameters so refer to their
+    respective documentation for a more complete list.
+
+    Parameters
+    ----------
+    config : `dict`, `StandardizerConfig` or `None`, optional
+        Collection of configuration key-value pairs.
+    kwargs : optional
+        Keyword arguments, assigned as configuration key-values.
+    """
+
+    def __init__(self, config=None, **kwargs):
+        # This is a bit hacky, but it makes life a lot easier because it
+        # enables automatic loading of the default configuration and separation
+        # of default config from instance bound config
+        keys = list(set(dir(self.__class__)) - set(dir(StandardizerConfig)))
```

Comment:
I'm a bit sketchy about the idea of adding logic to a configuration class (that doesn't affect IO but the config values). 

In general I can see why it's useful, but it tends to complicate things a great deal and can usually be refactored into the standardizer itself at a cost of 1 more flag. For example `do_masking` and `grow_mask`. We don't have a method setting `grow_mask` value if `do_masking` is set - we just ignore it. My initial reaction was that if there's a `compute_file_offset` that would be the job of a Standardizer, who would look up a `self.config.byte_offset` value if it's set and do a thing; but maybe I'm missing something. Could you explain a bit more what do you think as a "standardizer helper function"? Standardizers are, by definition, the helper functions that help translate data on disk to KBMOD abstraction - what parameters would we be wanting to make "derived params" [parameters depending on value of other parameters] for a Standardizer?

To answer your question though - yes, if you add a method to a subclass of `StandardizerConfig` this will add it to the `self._conf` dict. If you update, overwrite or replace it - this will replace it. In the list of `keys` this will appear as a string as you'd expect. In the list of `values` it would appear as a callable. If a `config` is provided that contains the same named key - it will be overridden with that value. This is all in line with what generally you can do in Python. 
In an image-collection this might be a problem, as we serialize all of the configs and we can't serialize a callable. So it wouldn't roundtrip. It is possible to filter out `"method"` from a stringified representation of attributes and get rid of it and expect it to exist on the roundtrip. This can be done here or in the image-collection.